vulnerabilities/api_v2.py

```diff
@@ -606,3 +610,116 @@ def get_queryset(self):
                 affected_package_vulnerability__vulnerability__vulnerability_id=vulnerability_id
             )
         return queryset
+
+
+class CreateListRetrieveUpdateViewSet(
+    mixins.CreateModelMixin,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    A viewset that provides `create`, `list, `retrieve`, and `update` actions.
+    To use it, override the class and set the `.queryset` and
+    `.serializer_class` attributes.
+    """
+
+    pass
+
+
+class PipelineRunAPISerializer(serializers.HyperlinkedModelSerializer):
+    status = serializers.SerializerMethodField()
+    execution_time = serializers.SerializerMethodField()
+    log = serializers.SerializerMethodField()
+
+    class Meta:
+        model = PipelineRun
+        fields = [
+            "run_id",
+            "status",
+            "execution_time",
+            "run_start_date",
+            "run_end_date",
+            "run_exitcode",
+            "run_output",
+            "created_date",
+            "vulnerablecode_version",
+            "vulnerablecode_commit",
+            "log",
+        ]
+
+    def get_status(self, run):
+        return run.status
+
+    def get_execution_time(self, run):
+        return round(run.execution_time, 2)
+
+    def get_log(self, run):
+        """Return only last 5000 character of log."""
+        return run.log[-5000:]
+
+
+class PipelineScheduleAPISerializer(serializers.HyperlinkedModelSerializer):
+    url = serializers.HyperlinkedIdentityField(
+        view_name="schedule-detail", lookup_field="pipeline_id"
+    )
+    latest_run = serializers.SerializerMethodField()
+    next_run_date = serializers.SerializerMethodField()
+
+    class Meta:
+        model = PipelineSchedule
+        fields = [
+            "url",
+            "pipeline_id",
+            "is_active",
+            "run_interval",
+            "created_date",
+            "schedule_work_id",
+            "next_run_date",
+            "latest_run",
+        ]
+
+    def get_next_run_date(self, schedule):
+        return schedule.next_run_date
+
+    def get_latest_run(self, schedule):
+        if latest := schedule.pipelineruns.first():
+            return PipelineRunAPISerializer(latest).data
+        return None
+
+
+class PipelineScheduleCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PipelineSchedule
+        fields = ["pipeline_id", "is_active", "run_interval"]
+        extra_kwargs = {
+            field: {"initial": PipelineSchedule._meta.get_field(field).get_default()}
+            for field in ["is_active", "run_interval"]
+        }
+
+
+class PipelineScheduleUpdateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PipelineSchedule
+        fields = ["is_active", "run_interval"]
+
+
+class PipelineScheduleV2ViewSet(CreateListRetrieveUpdateViewSet):
```

Comment:
IMO we should have tests to see how permissions work, test anon and admin users differently.

Reply:
Permission tests are here https://github.com/aboutcode-org/vulnerablecode/pull/1871/commits/9d8d22e2c47c03f0a5680abe3d262c8e54ddbaaa